rust/src/delta.rs

```diff
@@ -527,6 +572,10 @@ impl DeltaTable {
         }
     }
 
+    pub fn create_transaction(&mut self, options: Option<DeltaTransactionOptions>) -> DeltaTransaction {
```

Comment:
Yeah, I definitely get your opinion here. And since I already have a `pub new` on the `DeltaTransaction` I could just delete this helper. I don't have any huge master plan behind it, its just a common helper API I'm used to seeing in other contexts. 

I'll wait a bit before making the change in case anyone else wants to comment.

Reply:
Your mention @houqp is exactly what I was shooting for in the implementation - support either client code path. I am very friendly to @rtyler's suggestion as well though - abstractions can get leaky and cause problems down the road if they are not necessary and I do believe it is good to be very deliberate and explicit when defining your APIs, but I'm just so used to seeing this style of helper method elsewhere that it almost feels like a de facto standard helper method that should come out of the box. I don't have a super strong opinion myself regarding whether the helper should be included in the first commit myself. One thing worth mentioning is that it would be super easy to remove this helper for now and add it back later to be a bit "non-committal". Other than that, I will leave it to you guys to decide if this helper should be included in the initial API. Will watch for further comments.